packages/gatsby/src/utils/map-pages-to-static-query-hashes.ts

```diff
@@ -0,0 +1,174 @@
+import { uniqBy } from "lodash"
+import { IGatsbyState } from "../redux/types"
+
+const mapComponentsToStaticQueryHashes = (
+  staticQueryComponents: IGatsbyState["staticQueryComponents"]
+): Map<string, string> =>
+  Array.from(staticQueryComponents).reduce(
+    (map, [, { componentPath, hash }]) => {
+      map.set(componentPath, hash)
+      return map
+    },
+    new Map()
+  )
```

Comment:
sort of nit, but this could be more performant by just using Map.forEach and manually adding values to a parent scope map. This is technically doing 2 loops. But again, very nit.

Reply:
I agree, but mainly because it would be a lot more readable.

Reply:
Done